### DISCUSSION TITLE
The correct way to resume experiment with logger

### DISCUSSION THREAD

**Discussion by ashleve** [Lightning-AI/pytorch-lightning#5779] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
From what I understand checkpoints don't save id of the logger run right?
Then what is the correct way to resume training from checkpoint and resume the correct logger run at the same time?

For example, I'm using Wights&Biases logger where you can pass id argument which resumes run experiment:
```
WandbLogger(id=run_id)
```
How can I know what is the correct run id without checking it manually?
Is it possible to somehow store it in the checkpoint?
Also, why is checkpoint not storing things like that automatically? Seems like a useful feature.

**Comment by manifoldhiker** [comment]
Have the same question.

Maybe, `ModelCheckpoint` should also have an option to save wandb state as well?

**Comment by jonashaag** [comment]
Same question here.

**Comment by jonashaag** [comment]
Still an issue, stalebot.

**Reply by rlone2018** [reply]
I am also using Wights&Biases logger in PyTorch lightening.  Due to the limited GPU runtime in COLAB, I have to save my checkpoints and resume them in the next run. The problem is that my wandb graphs show discontinuity and I get separate lines instead of a straight one. Please help me to solve this issue.
![Screen Shot 2021-02-13 at 23 37 07](https://user-images.githubusercontent.com/58883533/107861091-70942f80-6e54-11eb-85ce-a3c7f4983e68.png)

**Reply by awaelchli** [reply]
Try to select `global_step` on the abcissa, maybe that helps

**Comment by awaelchli** [comment]
> Also, why is checkpoint not storing things like that automatically? Seems like a useful feature.

If you make a feature request on github, I will implement it.

**Reply by awaelchli** [reply]
You can also save it by overriding the `on_save_checkpoint` hook and dump the logger.version in there.

**Reply by ashleve** [reply]
@awaelchli feature request has been here for a while: https://github.com/PyTorchLightning/pytorch-lightning/issues/5342
I actually thought about implementing it myself but I'm too busy at the moment
